tests/components/recorder/test_history.py

```diff
@@ -265,12 +268,19 @@ def set_state(state):
     with patch(
         "homeassistant.components.recorder.core.dt_util.utcnow", return_value=point
     ):
-        states = [
-            set_state("idle"),
-            set_state("Netflix"),
-            set_state("Plex"),
-            set_state("YouTube"),
-        ]
+        states = [set_state("idle")]
+    with patch(
+        "homeassistant.components.recorder.core.dt_util.utcnow", return_value=point2
+    ):
+        states.append(set_state("Netflix"))
+    with patch(
+        "homeassistant.components.recorder.core.dt_util.utcnow", return_value=point3
+    ):
+        states.append(set_state("Plex"))
+    with patch(
+        "homeassistant.components.recorder.core.dt_util.utcnow", return_value=point4
+    ):
+        states.append(set_state("YouTube"))
```

Comment:
The behavior is different between MariaDB and SQLite: SQLite returns a list which is reverse sorted if the timestamps are exactly same. Maybe it implicitly reverse sorts also on the primary column?

I worked around it by making sure the events have differing time stamps. If we want the previous behavior to be consistent across engines we should perhaps sort also according to the id column, i.e. insertion order?